### PR TITLE
Specify operations on values typed as type variables

### LINKED ISSUE #1049: generic (specializable) functions and the expressions inside them
Maybe I missed it but I thought this would be valid but the reference compiler rejects it:
```
y func<y>(in y a)
{
  y b = a >> 1;
  return b;
}
```
I know in the specifications it say generic types can be bit types so I had assumed this would be valid so you could use func<bit<2>> later on. I didn't realize type checking for expressions was done before hand.

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
Signed-off-by: Mihai Budiu <mbudiu@vmware.com>
Fixes #1049

### CODE REVIEW

File: p4-16/spec/P4-16-spec.mdk [hunk 1]
```diff
@@ -4654,6 +4654,32 @@ bool b1 = (bit<32>)x == z;  // cast needed
...
+instantiated with specific type arguments.
+
+Currently the only operations that are available for such values are
+assignment (explicit through `=`, or implicit, through argument passing).
+This behavior is similar to languages such as Java, and different from
+languages such as C++.
+
+A future version of P4 may introduce a notion of _type constraints_
+which would enable more operations on such values.  Because of this
+limitaiton, currently such values of limited utility.
```
Comment (mbudiu-vmw) [inline p4-16/spec/P4-16-spec.mdk:4681]: limitation

Comment (jafingerhut) [inline p4-16/spec/P4-16-spec.mdk:4681]: suggested wordsmithing: "such values are currently of limited utility."

Comment (jeremias-blendin-intel) [reply p4-16/spec/P4-16-spec.mdk:4681]: There's a typo in last line: 'limitaiton' -> 'limitation'

**Review by jnfoster — APPROVED** [review]
LGTM

**Pull request merged**